frontend/components/Domain/Recipe/RecipeDialogTimeline.vue

```diff
@@ -0,0 +1,279 @@
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, ref, useAsync, useContext } from "@nuxtjs/composition-api";
+import { whenever } from "@vueuse/core";
+import RecipeTimelineContextMenu from "./RecipeTimelineContextMenu.vue";
+import { alert } from "~/composables/use-toast";
+import { useAsyncKey } from "~/composables/use-utils";
+import { useUserApi } from "~/composables/api";
+import { RecipeTimelineEventOut, RecipeTimelineEventUpdate } from "~/lib/api/types/recipe"
+import UserAvatar from "~/components/Domain/User/UserAvatar.vue";
+
+export default defineComponent({
+  components: { RecipeTimelineContextMenu, UserAvatar },
+
+  props: {
+    value: {
+      type: Boolean,
+      default: false,
+    },
+    slug: {
+      type: String,
+      default: "",
+    },
+    recipeName: {
+      type: String,
+      default: "",
+    },
+  },
+
+  setup(props, context) {
+    const api = useUserApi();
+    const { $globals, $vuetify, i18n } = useContext();
+    const timelineEvents = ref([{}] as RecipeTimelineEventOut[])
+
+    const useMobileFormat = computed(() => {
+      return $vuetify.breakpoint.smAndDown;
+    });
+
+    // V-Model Support
+    const dialog = computed({
+      get: () => {
+        return props.value;
+      },
+      set: (val) => {
+        context.emit("input", val);
+      },
+    });
+
+    whenever(
+      () => props.value,
+      () => {
+        refreshTimelineEvents();
+      }
+    );
+
+    function chooseEventIcon(event: RecipeTimelineEventOut) {
+      switch (event.eventType) {
+        case "comment":
+          return $globals.icons.commentTextMultiple;
+
+        case "info":
+          return $globals.icons.informationVariant;
+
+        case "system":
+          return $globals.icons.cog;
+
+        default:
+          return $globals.icons.informationVariant;
+      };
+    };
+
+    function updateTimelineEvent(index: number) {
+      const event = timelineEvents.value[index]
+      const payload: RecipeTimelineEventUpdate = {
+        subject: event.subject,
+        eventMessage: event.eventMessage,
+        image: event.image,
+      };
+
+      useAsync(async () => {
+        const { response } = await api.recipes.updateTimelineEvent(props.slug, event.id, payload);
+        if (response?.status !== 200) {
+          alert.error(i18n.t("events.something-went-wrong") as string);
+          return;
+        }
+
+        alert.success(i18n.t("events.event-updated") as string);
+      }, useAsyncKey());
+    }
+
+    function deleteTimelineEvent(index: number) {
+      useAsync(async () => {
+        const { response } = await api.recipes.deleteTimelineEvent(props.slug, timelineEvents.value[index].id);
+        if (response?.status !== 200) {
+          alert.error(i18n.t("events.something-went-wrong") as string);
+          return;
+        }
+
+        timelineEvents.value.splice(index, 1);
+        alert.success(i18n.t("events.event-deleted") as string);
+      }, useAsyncKey());
```

Comment:
Is there a reason you're using useAsync Here? That should only be necessary for top level statements.

Reply:
Oh duh, my Python brain assumed all functions were invoked synchronously, so I had to `await` outside of async. Instead I'll just make the function async. Thanks!